test/SwiftSyntax/AllTokenKindsInSyntaxGybSupport.c

```diff
@@ -0,0 +1,34 @@
+// RUN: %utils/gyb --line-directive '' %S/Inputs/TokenKindList.txt.gyb | sort > %T/python_kinds.txt
+// RUN: xcrun %clang %s -I %swift_src_root/Include -o %T/list_def_kinds.o
+// RUN: %T/list_def_kinds.o | sort > %T/def_kinds.txt
+// RUN: diff %T/def_kinds.txt %T/python_kinds.txt
+
+// Check that all token kinds listed in TokenKinds.def are also in
+// gyb_syntax_support/Token.py
+
+#include <stdio.h>
+#include <string.h>
+
+void printKeyword(char *name) {
+  // We don't expect any SIL related tokens on the SwiftSyntax side
+  if (!strcmp(name, "sil_dollar") ||
+      !strcmp(name, "sil_exclamation") ||
+      !strcmp(name, "sil_local_name")) {
+    return;
+  }
+
+  // These token kinds are internal only and should not be exposed on the
+  // SwiftSyntax side
+  if (!strcmp(name, "code_complete") ||
+      !strcmp(name, "comment") ||
+      !strcmp(name, "eof")) {
+    return;
+  }
+  printf("%s\n", name);
+}
+
+int main() {
+#define TOKEN(KW) printKeyword(#KW);
+#define SIL_KEYWORD(KW)
+#include "swift/Syntax/TokenKinds.def"
```

Comment:
Instead of adding another tool, we should add this print action into swift-ide-test; and printing the syntax token kinds in swift-swiftsyntax-test. The test should be a simple diff of two dumps.

Reply:
Thinking more about this, it's better if we put the print action into `swift-syntax-test`. It makes a lot of sense to compare dumps from `swift-syntax-test` and `swift-swiftsyntax-test`.

Reply:
I read `swift-syntax-test` anyway ;-)

Reply:
In case we keep the generated python list below it also makes sense to consider keeping this tool, since `swift-syntax-test` currently always operates on an input file and putting this into `swift-syntax-test` thus feels a little awkward.

Reply:
We can provide a dummy input file to `swift-syntax-test` for this action (I think it is totally OK). Drawbacks of having these standalone tools are (1) it's conventional to separate test driver and test data, like the way `swift-syntax-test` and `round_trip_parse_gen.swift` are separated; (2) we won't visit these tools for other purposes in the future very often, leading to the cognitive burden of figuring out why they are there in a longer term. So I still prefer we move these testing logics to a more popular tool.